org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1343,13 +1359,44 @@ protected void load (int ID, String trxName)
 		}
 	}	//	load
 
+	/**
+	 * Load record with UUID
+	 * 
+	 * @param uuID    UUID
+	 * @param trxName transaction name
+	 */
+	protected void loadByUU(String uuID, String trxName)
+	{
+		// reset new values
+		m_newValues = new Object[get_ColumnCount()];
+		checkImmutable();
+
+		if (log.isLoggable(Level.FINEST))
+			log.finest("uuID=" + uuID);
+		if (!Util.isEmpty(uuID, true))
+		{
+			load(uuID,trxName);
+		}else {
+			load(0,trxName);
+		}
+	} // loadByUU
```

Comment:
without the uuid constructor requirement, we just need to keep one of loadByUU(uuid, trxName) or load(uuid, trxName) (if we keep both, load(uuid, trxName) should be change to private or protected). and to keep things simple, for loadByUU(uuid, trxName), just return null if uuid is null or empty.

Reply:
Shouldn't we allow null UU to create new record?

Reply:
No, we don't need that  since the intention here is not to use uuid as primary key. The existing approach of id=0 should continue to be used to create new record.

Reply:
This also done in latest commit.